normandy/recipes/api/views.py

```diff
@@ -58,3 +58,26 @@ def retrieve(self, request, name, impl_hash):
             raise NotFound('Hash does not match current stored action.')
 
         return Response(action.implementation)
+
+
+class RecipeViewSet(viewsets.ModelViewSet):
+    """Viewset for viewing and uploading recipes."""
+    queryset = Recipe.objects.all()
+    serializer_class = RecipeSerializer
+    permission_classes = [
+        permissions.DjangoModelPermissions,
+        AdminEnabled,
+    ]
+
+    def update(self, request, *args, **kwargs):
+        """
+        Intercept PUT requests and have them create instead of update
+        if the object does not exist.
+        """
+        if request.method == 'PUT':
+            try:
+                self.get_object()
+            except Http404:
+                return self.create(request, *args, **kwargs)
```

Comment:
@Osmose I wonder if we should have a `normandy.base.api.ModelViewSet` that we can use as the base for our APIs that have this version of `update`, since I think we are going to want to do this on everything. (It also helps with the concerns I have on #109).

@rehandalal I don't think we need to make any changes to this PR about this.